### PR TITLE
[event] add subject page config to the pages

### PR DESCRIPTION
**Pull request by beets**
- Find the best place dcid for the config using recon. If not in a country, don't show charts.
- Add some basic charts for the place
- Updated return format of event properties (dict instead of serialized list) for ease of use

TODO: add more charts, and fallback logic if there isn't enough charts to display.

With charts:
image

### CODE REVIEW

**Review by chejennifer — COMMENTED** [review]
this is cool!

File: server/config/disaster_dashboard/events.textproto [hunk 1]
```diff
@@ -0,0 +1,189 @@
+# Copyright 2022 Google LLC
```
[the remark this answers is not in the document]
Comment (beets) [reply server/config/disaster_dashboard/events.textproto:1]: done

File: static/js/components/subject_page/category.tsx [hunk 2]
```diff
@@ -51,11 +52,14 @@ export const Category = memo(function Category(
...
+    place: props.place.name,
+    date: "",
+  };
+  const title = props.config.title ? formatString(props.config.title, rs) : "";
   return (
     <article className="category col-12" id={props.id}>
-      {props.config.title && (
-        <h2 className="block-title">{props.config.title}</h2>
-      )}
+      {props.config.title && <h2 className="block-title">{title}</h2>}
```
[the remark this answers is not in the document]
Comment (beets) [reply static/js/components/subject_page/category.tsx:62]: good point, done

File: static/js/apps/event/app.tsx [hunk 3]
```diff
@@ -45,19 +47,28 @@ const _IGNORED_PROPERTIES = new Set([
...
+const PAGE_ID = "event";
+
 /**
  * Stores information about the particular event node the event page should render.
  */
 interface AppPropsType {
   dcid: string;
   name: string;
-  properties: Array<Property>;
+  properties: Record<string, Property>;
```
[the remark this answers is not in the document]
Comment (beets) [reply static/js/apps/event/app.tsx:67]: good catch, done

File: static/js/apps/event/app.tsx [hunk 4]
```diff
@@ -137,25 +146,41 @@ export function App(props: AppPropsType): JSX.Element {
...
 }
 
 /**
  * Returns the first property in the list with any of the given dcids.
  */
-function findProperty(dcids: string[], properties: Array<Property>): Property {
-  return properties.find((p) => dcids.indexOf(p.dcid) >= 0);
+function findProperty(
+  dcids: string[],
+  properties: Record<string, Array<PropertyValue>>
```
[the remark this answers is not in the document]
Comment (beets) [reply static/js/apps/event/app.tsx:166]: good catch, done

File: static/js/apps/event/app.tsx [hunk 5]
```diff
@@ -137,25 +146,41 @@ export function App(props: AppPropsType): JSX.Element {
...
+): Array<PropertyValue> {
```
[the remark this answers is not in the document]
Comment (beets) [reply static/js/apps/event/app.tsx:167]: good catch, done

File: static/js/apps/event/app.tsx [hunk 6]
```diff
@@ -45,19 +47,28 @@ const _IGNORED_PROPERTIES = new Set([
...
   "irwinID",
   "wfigsFireID",
+  "affectedPlace",
   ..._LOCATION_PROPERTIES,
   ..._GEOJSON_PROPERTIES,
   ..._START_DATE_PROPERTIES,
   ..._END_DATE_PROPERTIES,
 ]);
 
+type Property = Array<PropertyValue>;
```
[the remark this answers is not in the document]
Comment (beets) [reply static/js/apps/event/app.tsx:57]: good point - updated to PropVals

File: server/routes/event.py [hunk 7]
```diff
@@ -46,12 +64,49 @@ def get_properties(dcid):
...
 
+def get_property_value(dcid: str, prop: str) -> str:
+  result = dc.property_values([dcid], prop)
+  if result[dcid]:
+    return result[dcid][0]
+  return None
+
+
+def get_places(properties) -> Dict[str, List[str]]:
+  """Returns { place_dcid: [place_types] } based on lat/long of event. """
```
[the remark this answers is not in the document]
Comment (beets) [reply server/routes/event.py:81]: done

File: server/routes/event.py [hunk 8]
```diff
@@ -46,12 +64,49 @@ def get_properties(dcid):
...
+  for prop, values in properties.items():
+    if prop in ['location', 'startLocation'] and len(values):
```
[the remark this answers is not in the document]
Comment (beets) [reply server/routes/event.py:83]: done

File: server/routes/event.py [hunk 9]
```diff
@@ -46,12 +64,49 @@ def get_properties(dcid):
...
+      dcid = values[0]['dcid']
```
[the remark this answers is not in the document]
Comment (beets) [reply server/routes/event.py:84]: added that

File: server/routes/event.py [hunk 10]
```diff
@@ -46,12 +64,49 @@ def get_properties(dcid):
...
+      coordinates = [{
+          'latitude': get_property_value(dcid, 'latitude'),
+          'longitude': get_property_value(dcid, 'longitude'),
+      }]
+      place_coordinates = dc.resolve_coordinates(coordinates).get(
```
[the remark this answers is not in the document]
Comment (beets) [reply server/routes/event.py:89]: yes, we should! done.

**Comment by beets** [comment]
thanks for the review! addressed your comments, ptal

**Review by chejennifer — APPROVED** [review]
thanks for the changes!

**Pull request merged**